src/architecture/unixfs.md

```diff
@@ -0,0 +1,618 @@
+  `[1, 999999999]` is **not** valid. This includes a fractional value of `0`.
+  Implementations encountering such values should consider the entire enclosing
+  metadata block malformed and abort processing the corresponding DAG.
+- The `mtime` structure is optional - its absence implies `unspecified`, rather
+  than `0`
+- For ergonomic reasons a surface API of an encoder MUST allow fractional 0 as
+  input, while at the same time MUST ensure it is stripped from the final structure
+  before encoding, satisfying the above constraints.
+
+Implementations interpreting the mtime metadata in order to apply it within a
+non-IPFS target MUST observe the following:
+
+- If the target supports a distinction between `unspecified` and `0`/`1970-01-01T00:00:00Z`,
+  the distinction must be preserved within the target. E.g. if no `mtime` structure
+  is available, a web gateway must **not** render a `Last-Modified:` header.
+- If the target requires an mtime ( e.g. a FUSE interface ) and no `mtime` is 
+  supplied OR the supplied `mtime` falls outside of the targets accepted range:
+  - When no `mtime` is specified or the resulting `UnixTime` is negative:
+    implementations must assume `0`/`1970-01-01T00:00:00Z` (note that such values
+    are not merely academic: e.g. the OpenVMS epoch is `1858-11-17T00:00:00Z`)
+  - When the resulting `UnixTime` is larger than the targets range ( e.g. 32bit
+    vs 64bit mismatch) implementations must assume the highest possible value
+    in the targets range (in most cases that would be `2038-01-19T03:14:07Z`)
+
+## Paths
+
+Paths first start with `<CID>/` or `/ipfs/<CID>/` where `<CID>` is a [multibase]
+encoded [CID]. The CID encoding MUST NOT use a multibase alphabet that have
+`/` (`0x2f`) unicode codepoints however CIDs may use a multibase encoding with
+a `/` in the alphabet if the encoded CID does not contain `/` once encoded.
+
+Everything following the CID is a collection of path component (some bytes)
+separated by `/` (`0x2F`). UnixFS paths read from left to right, and are
+inspired by POSIX paths.
+
+- Components MUST NOT contain `/` unicode codepoints because else it would break
+  the path into two components.
+- Components SHOULD be UTF8 unicode.
+- Components are case sensitive.
+
+### Escaping
+
+The `\` may be supposed to trigger an escape sequence. However, it is currently
+broken and inconsistent across implementations. Until we agree on a specification
+for this, you SHOULD NOT use any escape sequences and non-ASCII characters.
+
+### Relative Path Components
+
+Relative path components MUST be resolved before trying to work on the path:
+
+- `.` points to the current node, those path components MUST be removed.
+- `..` points to the parent node, they MUST be removed left to right. When removing
+  a `..`, the path component on the left MUST also be removed. If there is no path
+  component on the left, you MUST error since it is an attempt of out-of-bounds
+  path resolution.
```

Comment:
```suggestion
- `.` points to the current node and  MUST be removed.
- `..` points to the parent node and MUST be removed left to right. When removing
  a `..`, the path component on the left MUST also be removed. If there is no path
  component on the left, you MUST error to avoid out-of-bounds
  path resolution.
```